I93 bulkrax upgrade

ShanaLMoore opened the pull request:
REF: https://github.com/scientist-softserv/ams/issues/93

This PR upgrades bulkrax from 1.0.2 to 5.3.0 but leaves GBH's custom relationship code so that the XML importer can form child relations.

# XML PARSER
- [x] successfully imports data and forms/displays relationships (using their existing custom relationship code)

![image](https://github.com/WGBH-MLA/ams/assets/10081604/adfe0040-2593-4eb7-890b-9541fdb002f6)

![image](https://github.com/WGBH-MLA/ams/assets/10081604/4d25574a-8325-40ef-a689-4fe1e0b48696)

# XML/MANIFEST PARSER
Marked as low priority so we can make a separate ticket for this

# CSV PARSER
Marked as the lowest priority so we can make a separate ticket for this

Commit:
:construction: wip: upgrade bulkrax to the latest

upgrade bulkrax from v1.0.2 => v5.3.0

Issue:
- https://github.com/scientist-softserv/ams/issues/77

Commit:
Merge branch 'develop' into i77-upgrade-bulkrax

Commit:
:broom: add missing bulkrax migrations

Commit:
wip

Commit:
wip - resolve errors when running xml import

This commit includes the changes needed to successfully run an xml importer.

Commit:
remove byebugs

Commit:
:construction: work in progress

Figure out a way to bypass calls to the actor for assigning children to parent record.

Commit:
Pull in child_relationships_job

Job has been deleted from Bulkrax. This pulls the original file from
Bulkrax v1.0.0, and incorporates the overrides from the decorator file,
so the decorator file gets deleted.

There should be no behavior changes due to this pull commit.

Commit:
:broom: update method signature to be more in line with Bulkrax 5.3.0

I tested this with Bulkrax 1.0.2 and Bulkrax 5.3.0.

Commit:
:broom: add dual boot support for to the relationship job

add dual boot conditionals to handle the different attr name per bulkrax version

Commit:
Update pbcore_manifest_parser.rb

Commit:
Merge branch 'develop' into i93-bulkrax-upgrade

ShanaLMoore commented on app/views_rails_6_1/bulkrax/entries/show.html.old.erb:
```diff
@@ -1,4 +1,5 @@
 <%# OVERRIDE BULKRAX 1.0.2 to display ID instead of record class name %>
+<% rails 'hel' %>
```
oops

Commit:
:broom: cleanup PR

Reverts some unnneccessary changes

Commit:
Merge branch 'i93-bulkrax-upgrade' of https://github.com/WGBH-MLA/ams into i93-bulkrax-upgrade

Commit:
:broom: remove unused views

Commit:
:lipstick: add code comment to justify decorator

Commit:
:broom: remove dual boot conditionals

when we run migrations some attrs will not be available so it doesn't matter which rails version we boot.

kirkkwang commented on Gemfile:
```diff
@@ -21,8 +21,7 @@ if ENV['DEPENDENCIES_NEXT'] && !ENV['DEPENDENCIES_NEXT'].empty?
   gem 'twitter-typeahead-rails', '0.11.1.pre.corejavascript'
   group :bulkrax do
```
i'm thinking we don't need this group anymore since it's just bulkrax now

ShanaLMoore replied:
true

ShanaLMoore commented:
converting back to draft to resolve failing 5.1 spec. the 6.1 specs can be handled in their respective tickets.

Commit:
:broom: remove group wrapper around bulkrax and make private method to public

This commit addresses PR feedback and resolves the failing spec in the rails 5.1 context.

Commit:
:broom: collection_field is no longer a valid method

This commit fixes the failing spec for rails 5.1

Commit:
:broom: skip failing csv entry specs

We are aware the the csv parser is currently failing. Because it's the last priority, we will skip these specs for now and resolve them when we work on the CSV parser.

ref: https://github.com/scientist-softserv/ams/issues/105
